keps/sig-storage/20181111-extend-datasource-field.md

```diff
@@ -114,6 +115,8 @@ Currently the CSI provisioner already accepts the DataSource field in new provis
 
 To emphasize above, this feature will ONLY be available for CSI.  This feature wil NOT be added to In-tree plugins or Flex drivers, this is strictly a CSI only feature.
 
+It's important to note that we intentionally require that a source PVC be in the same StorageClass as the PVC being created.  This is currently required because the StorageClass determintes characteristics for a volume like `fsType`.  Performing a clone from an xfs volume to an ext4 volume for example would not be acceptable, given that a storageClass can have unique information, cloning across storage classes is not something we're able to try and determine safely at this time.
```

Comment:
It's fine for beta, but we should reassess this requirement for GA and consider loosening it: Instead of adding this limitation/validation to k8s, maybe we say it is up to the storage system to decide if it can support the clone. If someone tries to clone and old/new StorageClass don't match, we can leave it up to the storage system to make that decision and fail the CreateVolume call if it wants to?

Reply:
@saad-ali Working on update to GA status and was going to move forward with rewording/lifting the StorageClass restriction.  This seems reasonable to me (still), just want to make sure no new thoughts here since July on your side.

Reply:
@msau42 ^^

Reply:
I'm not sure we want to lift the StorageClass restriction. We need to very carefully consider the implications given https://github.com/kubernetes/kubernetes/issues/85233